felix/bpf/proxy/syncer.go

```diff
@@ -542,13 +550,24 @@ func (s *Syncer) apply(state DPSyncerState) error {
 
 	// insert or update existing services
 	for sname, sinfo := range state.SvcMap {
+		hintsAnnotation := sinfo.HintsAnnotation()
+
 		log.WithField("service", sname).Debug("Applying service")
 		skey := getSvcKey(sname, "")
 
 		eps := make([]k8sp.Endpoint, 0, len(state.EpsMap[sname]))
 		for _, ep := range state.EpsMap[sname] {
+			zoneHints := ep.GetZoneHints()
 			if ep.IsReady() {
-				eps = append(eps, ep)
+				if ShouldAppendTopologyAwareEndpoint(nodeLabels, hintsAnnotation, zoneHints) {
```

Comment:
This works, just wondering if it would be a little cleaner/easier to do it in proxy.go before Apply() gets called so that the syncer really get just eps that are valid. It could also filter our the ep.IsReady() in the same vein. Bu no strong opinion on this :)

Reply:
Thanks @tomastigera - I could investigate this option but would like to expedite the current PR now